content/posts/2021/reana-client-0.7.3.md

```diff
@@ -0,0 +1,346 @@
+
+```diff
+       - name: run-script
+         environment: 'python:3.9-slim'
+         commands:
+-          - python "${script_path}" "${sleeptime}"
++          - sudo python "${script_path}" "${sleeptime}"
+```
+
+Let us run the validator:
+
+```console
+$ reana-client validate -f reana.yaml
+==> Verifying REANA specification file... my-analysis/reana.yaml
+  -> SUCCESS: Valid REANA specification file.
+==> Verifying workflow parameters and commands...
+  -> WARNING: Operation "sudo" found in step "run-script" might be dangerous.
+```
+
+The output shows a warning message indicating the dangerous operation
+and in which step was found.
+
+### Workflow environment images validation
+
+Workflows run in containerised environments that can evolve
+throughout versions via tags. We encourage users to tag their images
+when running analyses as this will ensure reproducibility in the
+future, one of the pillars of
+[FAIR principles](https://www.nature.com/articles/sdata201618).
+
+Besides, it is important to verify the existence of these
+environment images, both locally and remotely (Docker Hub and
+GitLab registry), to ensure that the REANA cluster can pull them with
+no issues.
+
+This is why we are introducing a new option for the `validate`
+command, called `--environments`. When passing this option to the
+validator, we can perform the aforementioned verifications.
+
+Let us use the previous workflow example to depict this
+functionality. In that workflow, the environment used was
+`python:3.9-slim`, which is properly tagged and exists in
+Docker Hub. We are going to modify it and use a non-existing image
+instead, for example `foo:bar`.
+
+```diff
+   specification:
+     steps:
+       - name: run-script
+-        environment: 'python:3.9-slim'
++        environment: 'foo:bar'
+         commands:
+           - python "${script_path}" "${sleeptime}"
+```
+
+Let us call the validator, passing the environments option:
+
+```console
+$ reana-client validate -f reana.yaml --environments
+==> Verifying REANA specification file... my-analysis/reana.yaml
+  -> SUCCESS: Valid REANA specification file.
+==> Verifying workflow parameters and commands...
+==> Verifying environments in REANA specification file...
+  -> SUCCESS: Environment image foo:bar has the correct format.
+  -> WARNING: Environment image foo:bar does not exist locally.
+  -> WARNING: Environment image foo:bar does not exist in Docker Hub: "Resource not found"
+  -> ERROR: Environment image foo:bar does not exist locally or remotely.
+```
+
+Let us analyse the environment validation output. First, we get a
+success message informing us that the image has the correct format.
+This is because, as expected, it has the image name and the tag, so
+in that regard it is correct. Then we get two warnings telling us
+that the image was not found either locally nor in the Docker Hub
+registry.
+As a consequence of this, we can see the last error message, telling
```

Comment:
As a consequence, we see the validation ending with an error message, telling us that the image does not exist, and the validation failsl.